nansat/mappers/mapper_sentinel1_l1.py

```diff
@@ -415,125 +368,288 @@ def __init__(self, filename, gdalDataset, gdalMetadata,
+        data['longitude'] = new_lon.reshape(data['shape'])
+        data['height'] = new_height
+        return data
+
+    @staticmethod
+    def create_gcps(x, y, z, p, l):
+        """ Create GCPs from geolocation data
+
+        Parameters
+        ----------
+        x, y, z, p, l
+        N-D arrays with vakye of X, Y, Z, Pixel and Line coordinates.
+        X and Y are typically lon, lat.
+
+        Returns
+        -------
+        gcps : list with GDAL GCPs
+
+        """
+        gcps = []
+        for xi, yi, zi, pi, li in zip(x.flat, y.flat, z.flat, p.flat, l.flat):
+            gcps.append(gdal.GCP(xi, yi, zi, pi, li))
+        return gcps
+
+    def read_manifest_data(self, input_file):
+        """ Read information (time_coverage_start, etc) manifest XML
+
+        Parameters:
+        ----------_
+        input_file : str
+            name of manifest file
+
+        Returns:
+        --------
+        data : dict
+            manifest data. Keys:
+                time_coverage_start
+                time_coverage_end
+                platform_familyName
+                platform_number
+
+        """
+
+        data = {}
+        xml = self.read_vsi(input_file)
+        # set time as acquisition start time
+        n = Node.create(xml)
+        meta = n.node('metadataSection')
+        for nn in meta.children:
+            if str(nn.getAttribute('ID')) == 'acquisitionPeriod':
+                # get valid time
+                data['time_coverage_start'] = parse((nn.node('metadataWrap').
+                                                     node('xmlData').
+                                                     node('safe:acquisitionPeriod')['safe:startTime']
+                                                     )).isoformat()
+                data['time_coverage_end'] = parse((nn.node('metadataWrap').
+                                                   node('xmlData').
+                                                   node('safe:acquisitionPeriod')['safe:stopTime']
+                                                   )).isoformat()
+            if str(nn.getAttribute('ID')) == 'platform':
+                data['platform_familyName'] = str(nn.node('metadataWrap').
+                                                     node('xmlData').
+                                                     node('safe:platform')['safe:familyName'])
+                data['platform_number'] = str(nn.node('metadataWrap').
+                                                     node('xmlData').
+                                                     node('safe:platform')['safe:number'])
+        return data
+
+    def vrts_from_arrays(self, data, variable_names, pol='', resize=True, resample_alg=2):
+        """ Convert input dict with arrays into dict with VRTs
+
+        Parameters:
+        ----------_
+        data : dict with 2D arrays
+        variable_names : list of variable names
+        pol : str (HH, ...)
```

Comment:
shouldn't explanations come on the next line?

Reply:
Fixed here and below